src/EA.Weee.Integration.Tests/Handlers/GetSchemesWithObligationHandlerTests.cs

```diff
@@ -0,0 +1,121 @@
+            {
+                LocalSetup();
+
+                var authority1 = Query.GetCompetentAuthorityByName("Environment Agency");
+                var authority2 = Query.GetCompetentAuthorityByName("Northern Ireland Environment Agency");
+                var organisation = OrganisationDbSetup.Init().Create();
+                scheme1 = SchemeDbSetup.Init()
+                    .WithAuthority(authority1.Id)
+                    .WithOrganisation(organisation.Id)
+                    .WithStatus(SchemeStatus.Approved)
+                    .Create();
+
+                scheme2 = SchemeDbSetup.Init()
+                    .WithAuthority(authority2.Id)
+                    .WithOrganisation(organisation.Id)
+                    .WithStatus(SchemeStatus.Approved)
+                    .Create();
+
+                schemeNotMatchingComplianceYear = SchemeDbSetup.Init()
+                    .WithAuthority(authority1.Id)
+                    .WithOrganisation(organisation.Id)
+                    .WithStatus(SchemeStatus.Approved)
+                    .Create();
+
+                var obligationUploadCurrentYear = ObligationUploadDbSetup.Init().Create();
+                var obligationUploadNotMatchingYear = ObligationUploadDbSetup.Init().Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(scheme1.Id)
+                    .WithObligationUpload(obligationUploadCurrentYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(scheme2.Id)
+                    .WithObligationUpload(obligationUploadCurrentYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(schemeNotMatchingComplianceYear.Id)
+                    .WithObligationUpload(obligationUploadNotMatchingYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year - 1)
+                    .Create();
+
+                ObligationSchemeDbSetup.Init().WithScheme(schemeNotMatchingComplianceYear.Id)
+                    .WithObligationUpload(obligationUploadNotMatchingYear.Id)
+                    .WithComplianceYear(SystemTime.UtcNow.Year + 1)
+                    .Create();
+
+                request = new GetSchemesWithObligation(SystemTime.UtcNow.Year);
+            };
+
+            private readonly Because of = () =>
+            {
+                result = Task.Run(async () => await handler.HandleAsync(request)).Result;
+            };
+
+            private readonly It shouldHaveReturnedObligationYears = () =>
+            {
+                result.Should().NotBeNull();
+                result.Count.Should().Be(2);
+                result.Should().Contain(s => s.Id == scheme1.Id);
+                result.Should().Contain(s => s.Id == scheme2.Id);
+                result.Should().OnlyHaveUniqueItems();
```

Comment:
You could add result.Should().NotContain for the schemeNotMatchingComplianceYear